src/test/java/it/geosolutions/httpproxy/HttpProxyTest.java

```diff
@@ -20,96 +20,261 @@
+import org.apache.http.message.BasicHeader;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.mockito.Matchers;
 import org.mockito.Mockito;
 
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
 /**
  * HttpProxyTest class. Test Cases for the HTTPProxy servlet.
- * 
+ *
  * @author Lorenzo Natali at lorenzo.natali@geo-solutions.it
  */
 public class HttpProxyTest extends Mockito {
-	final ServletConfig servletConfig = mock(ServletConfig.class);
-	ServletContext ctx = mock(ServletContext.class);
-	Map<String, String[]> parameters = new HashMap<String, String[]>();
-	private List<Header> headers = new ArrayList<Header>();
-
-	@Before
-	public void setUp() {
-		File f = new File(getClass().getClassLoader()
-				.getResource("test-proxy.properties").getFile());
-		when(ctx.getInitParameter("proxyPropPath")).thenReturn(
-				f.getAbsolutePath());
-		when(servletConfig.getServletContext()).thenReturn(ctx);
-
-		// setup base parameters
-
-		parameters.put("url", new String[] { "http://sample.com/" });
-
-	}
-
-	@Test
-	public void testRedirectGet() throws Exception {
-		HttpClient mockHttpClient;
-		HTTPProxy proxy;
-
-		mockHttpClient = mock(HttpClient.class);
-
-		// mock redirect response
-		final GetMethod mockGetMethod = mock(GetMethod.class);
-		when(mockHttpClient.executeMethod(mockGetMethod)).thenReturn(302);
-		String fakeLocation = "http://newURL.com/";
-		when(mockGetMethod.getResponseHeader(Utils.LOCATION_HEADER))
-				.thenReturn(new Header("Location", fakeLocation));
-
-		proxy = new HTTPProxy() {
-			private static final long serialVersionUID = 1L;
-
-			@Override
-			public GetMethod getGetMethod(URL url) {
-				return mockGetMethod;
-			}
-		};
-		proxy.setHttpClient(mockHttpClient);
-		proxy.init(servletConfig);
-
-		// mock http request to proxy
-		HttpServletRequest getRequest = mock(HttpServletRequest.class);
-		when(getRequest.getParameterMap()).thenReturn(parameters);
-		when(getRequest.getHeaderNames()).thenReturn(
-				Collections.enumeration(headers));
-		when(getRequest.getRequestURL()).thenReturn(
-				new StringBuffer("http://proxy.com/http-proxy/proxy"));
-		// mock http response object
-		HttpServletResponse getResponse = mock(HttpServletResponse.class);
-		final StubServletOutputStream servletOutputStream = new StubServletOutputStream();
-		when(getResponse.getOutputStream()).thenReturn(servletOutputStream);
-
-		proxy.doGet(getRequest, getResponse);
-		verify(getResponse).sendRedirect(
-				"http://proxy.com/http-proxy/proxy?url="
-						+ URLEncoder.encode(fakeLocation, "UTF-8"));
-		final byte[] data = servletOutputStream.baos.toByteArray();
-		Assert.assertNotNull(data);
-		Assert.assertTrue(data.length == 0);
-	}
+
+    private final int localPort = 8080;
+
+    @Rule
+    public WireMockRule wireMockRule =
+            new WireMockRule(WireMockConfiguration.options().port(5555));
+
+    @Rule
+    public WireMockRule wireMockProxyRule =
```

Comment:
I understand that you are expecting the HTTP wire to mock to be running, but it should be started (and stopped) by the test in a transparent way. This should be the way GeoTools is using it.